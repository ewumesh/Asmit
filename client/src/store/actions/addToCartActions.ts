// - Import action types
import { AddToCartActionType } from 'constants/addToCartActionType'
import { Cart, Order } from 'src/core/domain/cart'
import { ICartService } from 'src/core/services/cart'
import { SocialError } from 'src/core/domain/common'
import { SocialProviderTypes } from 'src/core/socialProviderTypes'
import { provider } from 'src/ecommerceEngine'
import * as globalActions from 'store/actions/globalActions'

/**
 * Get service providers
 */
const cartService: ICartService = provider.get<ICartService>(SocialProviderTypes.CartService)

export const dbAddProductToCart = (cartId: string, cart: Cart) => {
  return (dispatch: Function, getState: Function) => {

    return cartService.addCartItem(cartId, cart).then((result) => {
      cart.itemId = result.itemId
      dispatch(addProductToCart(cartId,cart))
    })
      .catch((error: SocialError) => dispatch(globalActions.showMessage(error.code)))
  }

}

export const dbUpdateCartItem = (itemId: string, cartId: string, cart: Cart) => {
  return (dispatch: Function, getState: Function) => {

    return cartService.updateCartItem(itemId,cartId, cart).then((result) => {
      if ( result.quantity === 0 ) {
          dispatch(removeProductFromCart(cartId,cart))
      } else {
          dispatch(updateProductToCart(cartId,cart))
      }

    })
      .catch((error: SocialError) => dispatch(globalActions.showMessage(error.code)))
  }

}

export const dbAddOrder = (order: Order) => {
  return (dispatch: Function, getState: Function) => {
    return cartService.addOrder(order).then((result: Order ) => {
      dispatch(addOrderToCart(result))
    })
      .catch((error: SocialError) => dispatch(globalActions.showMessage(error.code)))
  }

}

export const dbUpdateOrder = (orderId: string, order: Order) => {
  return (dispatch: Function, getState: Function) => {
    return cartService.updateOrder(orderId, order).then((result: Order ) => {
        dispatch(clearCart())
    })
      .catch((error: SocialError) => dispatch(globalActions.showMessage(error.code)))
  }

}
  /* _____________ CRUD State _____________ */

  /**
   * Add a list of rigions
   */
  export const addProductToCart = (cartId: string, cart: Cart) => {

    return {
      type: AddToCartActionType.PRODUCT_TO_CART,
      payload: {cartId, cart}
    }
  }

  /**
   * Add a list of rigions
   */
  export const updateProductToCart = (cartId: string, cart: Cart) => {

    return {
      type: AddToCartActionType.UPDATE_PRODUCT_TO_CART,
      payload: {cartId, cart}
    }
  }

  /**
   * Add a list of rigions
   */
  export const removeProductFromCart = (cartId: string, cart: Cart) => {

    return {
      type: AddToCartActionType.REMOVE_PRODUCT_FROM_CART,
      payload: {cartId, cart}
    }
  }

  /**
   * Add a list of rigions
   */
  export const addOrderToCart = (order: Order) => {

    return {
      type: AddToCartActionType.ORDER_TO_CART,
      payload: {order}
    }
  }

  /**
   * Clea search list in product store
   */
  export const clearCart = () => {
    return {
      type: AddToCartActionType.CLEAR_ALL_DATA_CART
    }
  }
