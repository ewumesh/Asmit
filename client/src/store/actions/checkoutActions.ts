// - Import action types
import { CheckoutActionType } from 'constants/checkoutActionType'

  /* _____________ CRUD State _____________ */

  /**
   * Add a list of rigions
   */
  export const setActiveStep = (step: number) => {
    console.log('checkout comes in checkout action')
    return {
      type: CheckoutActionType.ACTIVE_STEP,
      payload: {step}
    }
  }
