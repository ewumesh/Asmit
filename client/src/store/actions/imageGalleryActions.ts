// - Import react componetns
import { Image } from 'src/core/domain/imageGallery'
import { ImageGalleryActionType } from 'constants/imageGalleryActionType'

/* _____________ CRUD State _____________ */

/**
 * Add image list to image gallery
 * @param {Image[]} images is an array of images
 */
export const addImageList = (images: Image[]) => {
  return { type: ImageGalleryActionType.ADD_IMAGE_LIST_GALLERY,payload: images }
}

/**
 * Add image to image gallery
 * @param {Image} image
 */
export const addImage = (image: Image) => {
  return { type: ImageGalleryActionType.ADD_IMAGE_GALLERY, payload: image }
}

/**
 * Delete an image
 * @param  {string} id is an image identifier
 */
export const deleteImage = (id: string) => {
  return { type: ImageGalleryActionType.DELETE_IMAGE, payload: id }

}

/**
 * Delete an image
 */
export const setImageURL = (name: string, url: string) => {
  return {
    type: ImageGalleryActionType.SET_IMAGE_URL,
    payload: { name, url }
  }

}

/**
 * Clear all data in image gallery store
 */
export const clearAllData = () => {
  return {
    type: ImageGalleryActionType.CLEAT_ALL_DATA_IMAGE_GALLERY
  }
}

/**
 * Send image request
 */
export const sendImageRequest = (name: string) => {
  return {
    type: ImageGalleryActionType.SEND_IMAGE_REQUEST,
    payload: name
  }

}
