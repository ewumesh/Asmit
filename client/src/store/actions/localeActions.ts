import { setActiveLanguage } from 'react-localize-redux'

import { LanguageType } from 'store/reducers/locale/langugeType'

/**
 *  Set active language for translation
 */
export const setLanguage = (language: LanguageType) => {
  return (dispatch: Function , getState: Function) => {

    // Dispatch `setActiveLanguage` and pass the language.
    dispatch(setActiveLanguage(language))
  }
}
