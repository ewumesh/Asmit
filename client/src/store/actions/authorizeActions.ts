// - Import react components
import { push } from 'connected-react-router'

import { AuthorizeActionType } from 'constants/authorizeActionType'
import { IAuthorizeService } from 'src/core/services/authorize'
import { OAuthType, LoginUser } from 'src/core/domain/authorize'
import { SocialError } from 'src/core/domain/common'
import { SocialProviderTypes } from 'src/core/socialProviderTypes'
import { User } from 'src/core/domain/users'
import { UserRegisterModel } from 'src/models/users/userRegisterModel'
import { provider } from 'src/ecommerceEngine'
import * as globalActions from 'store/actions/globalActions'

/* _____________ CRUD State _____________ */

/**
 * Loing user
 * @param {string} uids
 */
export const login = (uid: string, email: string, displayName: string, authed: boolean) => {
  return {
    type: AuthorizeActionType.LOGIN,
    payload: { authed, email, uid, displayName }
  }
}

/**
 * Logout user
 */
export const logout = (msg: string, status: boolean) => {
  return {
    type: AuthorizeActionType.LOGOUT,
    payload: { msg, status }
  }
}

/**
 * User registeration call
 * @param user  for registering
 */
export const signup = (user: UserRegisterModel) => {
  return {
    type: AuthorizeActionType.SIGNUP,
    payload: { ...user }
  }

}

/**
 * Update user's password
 */
export const updatePassword = () => {
  return { type: AuthorizeActionType.UPDATE_PASSWORD }
}

/**
 * Get service providers
 */
const authorizeService: IAuthorizeService = provider.get<IAuthorizeService>(SocialProviderTypes.AuthorizeService)

/* _____________ CRUD DB _____________ */

/**
 * Log in user in server
 */
export const dbLogin = (email: string, password: string) => {
  return (dispatch: any, getState: any) => {
    return authorizeService.login(email, password).then((result) => {
      dispatch(login(result.uid, result.email,result.displayName, result.uid ? true : false))
      dispatch(push('/'))
    }, (error: SocialError) => {
        dispatch(globalActions.showMessage(error.message))
      }
    )
  }
}

/**
 * Log out user in server
 */
export const dbLogout = () => {
  return (dispatch: any, getState: any) => {
    return authorizeService.logout().then((result) => {
      dispatch(logout(result.msg, result.status))
      dispatch(push('/'))
    }, (error: SocialError) => dispatch(globalActions.showMessage(error.code)))
  }

}

/**
 * Send email verification
 */
export const dbSendEmailVerfication = () => {
  return (dispatch: any, getState: any) => {
    dispatch(globalActions.showNotificationRequest())

    return authorizeService.sendEmailVerification().then(() => {
      // Send email verification successful.
      dispatch(globalActions.showNotificationSuccess())
      dispatch(push('/'))
    })
      .catch((error: SocialError) => {
        // An error happened.
        dispatch(globalActions.showMessage(error.code))

      })
  }
}

/**
 *
 * @param user for registering
 */
export const dbSignup = (user: UserRegisterModel) => {
  return (dispatch: Function, getState: Function) => {
    let newUser = new User()
    newUser.email = user.email
    newUser.password = user.password
    newUser.fullName = user.fullName

    return authorizeService.registerUser(newUser).then((result) => {
      dispatch(signup({
        userId: result.uid,
        fullName: result.name,
        email: result.email,
        ...user
      }))
      dispatch(dbSendEmailVerfication())
      dispatch(push('/emailVerification'))
    })
      .catch((error: SocialError) => dispatch(globalActions.showMessage(error.code)))
  }

}

/**
 * Change user's password
 * @param {string} newPassword
 */
export const dbUpdatePassword = (newPassword: string) => {
  return (dispatch: any, getState: any) => {
    dispatch(globalActions.showNotificationRequest())

    return authorizeService.updatePassword(newPassword).then(() => {

      // Update successful.
      dispatch(globalActions.showNotificationSuccess())
      dispatch(updatePassword())
      dispatch(push('/'))
    })
      .catch((error: SocialError) => {
        // An error happened.
        switch (error.code) {
          case 'auth/requires-recent-login':
            dispatch(globalActions.showMessage(error.code))
            dispatch(dbLogout())
            break
          default:

        }
      })
  }
}

/**
 * Reset user's password
 * @param {string} newPassword
 */
export const dbResetPassword = (email: string) => {
  return (dispatch: any, getState: any) => {
    dispatch(globalActions.showNotificationRequest())

    return authorizeService.resetPassword(email).then(() => {

      // Reset password successful.
      dispatch(globalActions.showNotificationSuccess())
      dispatch(push('/login'))
    })
      .catch((error: SocialError) => {
        // An error happened.
        dispatch(globalActions.showMessage(error.code))

      })
  }
}

/**
 * Login user with OAuth
 */
export const dbLoginWithOAuth = (type: OAuthType) => {
  return (dispatch: any, getState: any) => {
    dispatch(globalActions.showNotificationRequest())

    return authorizeService.loginWithOAuth(type).then((result: LoginUser) => {
      // Send email verification successful.
      dispatch(globalActions.showNotificationSuccess())
      dispatch(login(result.uid, result.email, result.displayName, result.uid ? true : false))
      dispatch(push('/'))
    })
      .catch((error: SocialError) => {
        // An error happened.
        dispatch(globalActions.showMessage(error.code))

      })
  }
}
