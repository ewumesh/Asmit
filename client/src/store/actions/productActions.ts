// - Import domain
import { Map, fromJS } from 'immutable'
import { IProductService } from 'src/core/services/products'
import { Product } from 'src/core/domain/products'
import { ProductActionType } from 'constants/productActionType'
import { SocialError } from 'src/core/domain/common'
import { SocialProviderTypes } from 'src/core/socialProviderTypes'
import { provider } from 'src/ecommerceEngine'
import * as globalActions from 'store/actions/globalActions'

/**
 * Get service providers
 */
const productService: IProductService = provider.get<IProductService>(SocialProviderTypes.ProductService)

/**
 * Get all user products from data base
 */
export const dbGetProducts = (page: number = 0, limit: number = 10) => {
  return (dispatch: any, getState: Function) => {
    const lastPageRequest = 10
    if (lastPageRequest !== page) {
      return productService.getProducts(page, limit).then((result) => {
        if (!result.products || !(result.products.length > 0)) {
          return dispatch(notMoreDataStream())
        }

        let parsedData: Map<string, Map<string, any>> = Map({})
        result.products.forEach((product) => {
          const productId = Object.keys(product)[0]
          const productData = product[productId]
          parsedData = parsedData.setIn([productId], fromJS(productData))
        })
        dispatch(productsPageCount(result.pageCount))
        dispatch(addProducts(parsedData))
      })
        .catch((error: SocialError) => {
          dispatch(globalActions.showMessage(error.message))
        })

    }
  }
}

/**
 * Get all user products by search text from data base
 */
export const dbGetProductsBySearchText = (page: number = 0, limit: number = 10, searchText: string) => {
  return (dispatch: any, getState: Function) => {
    const lastPageRequest = 10
    if (lastPageRequest !== page) {
      return productService.getProductsBySearchText(page, limit, searchText).then((result) => {
        if (!result.products || !(result.products.length > 0)) {
          return dispatch(notMoreDataStream())
        }

        let parsedData: Map<string, Map<string, any>> = Map({})
        result.products.forEach((product) => {
          const productId = Object.keys(product)[0]
          const productData = product[productId]
          parsedData = parsedData.setIn([productId], fromJS(productData))
        })
        dispatch(productsPageCount(result.pageCount))
        dispatch(addProductsBySearchText(parsedData))
      })
        .catch((error: SocialError) => {
          dispatch(globalActions.showMessage(error.message))
        })

    }
  }
}

/**
 * Get all user products by category from data base
 */
export const dbGetProductsByCategory = (page: number = 0, limit: number = 10, category: string) => {
  return (dispatch: any, getState: Function) => {
    const lastPageRequest = 10
    if (lastPageRequest !== page) {
      return productService.getProductsByCategory(page, limit, category).then((result) => {
        if (!result.products || !(result.products.length > 0)) {
          return dispatch(notMoreDataStream())
        }

        let parsedData: Map<string, Map<string, any>> = Map({})
        result.products.forEach((product) => {
          const productId = Object.keys(product)[0]
          const productData = product[productId]
          parsedData = parsedData.setIn([productId], fromJS(productData))
        })
        dispatch(productsPageCount(result.pageCount))
        dispatch(addProductsByCategory(parsedData))
      })
        .catch((error: SocialError) => {
          dispatch(globalActions.showMessage(error.message))
        })

    }
  }
}

/**
 * Get all user products by category from data base
 */
export const dbGetProductsAttributes = (productId: string) => {
  return (dispatch: any, getState: Function) => {
      return productService.getProductAttributes(productId).then((result) => {
        if (!result.products || !(result.products.length > 0)) {
          return dispatch(notMoreDataStream())
        }

        let parsedData: Map<string, Map<string, any>> = Map({})
        result.products.forEach((product) => {
          const productId = Object.keys(product)[0]
          const productData = product[productId]
          parsedData = parsedData.setIn([productId], fromJS(productData))
        })

        dispatch(addProductsAttributes(parsedData))
      })
        .catch((error: SocialError) => {
          dispatch(globalActions.showMessage(error.message))
        })
  }
}

/**
 * Get all product details
 */
export const dbGetProductById = ( productId: string ) => {
  return (dispatch: any, getState: Function) => {

      return productService.getProductById(productId).then((product: Product) => {
        console.log('product detail in the product action --- ' + product )
        dispatch(addProduct(product))
      })
        .catch((error: SocialError) => {
          dispatch(globalActions.showMessage(error.message))
        })
  }
}

/* _____________ CRUD State _____________ */

/**
 * Add a normal product
 */
export const addProduct = (product: Product) => {
  console.log('product detail in the addProduct in product action --- ' + product )
  return {
    type: ProductActionType.ADD_PRODUCT_DETAIL,
    payload: { product }
  }
}

/**
 * Update a product
 */
export const updateProduct = (product: Map<string, any>) => {
  return {
    type: ProductActionType.UPDATE_PRODUCT,
    payload: { product }
  }
}

/**
 * Update the comments of product
 */
export const updateProductComments = (comments: Map<string, any>) => {
  return {
    type: ProductActionType.UPDATE_PRODUCT,
    payload: comments
  }
}

/**
 * Update the votes of product
 */
export const updateProductVotes = (votes: Map<string, any>) => {
  return {
    type: ProductActionType.UPDATE_PRODUCT,
    payload: votes
  }
}

/**
 * Delete a product
 */
export const deleteProduct = (uid: string, id: string) => {
  return {
    type: ProductActionType.DELETE_PRODUCT,
    payload: { uid, id }
  }
}

/**
 *  Products Page Count
 */
export const productsPageCount = (productsPageCount: number) => {
  return {
    type: ProductActionType.PRODUCTS_PAGE_COUNT,
    payload: { productsPageCount }
  }
}

/**
 * Add a list of product
 */
export const addProducts = (userProducts: Map<string, Map<string, any>>) => {
  return {
    type: ProductActionType.ADD_LIST_PRODUCT,
    payload: { userProducts }
  }
}

/**
 * Add a list of product by Search text
 */
export const addProductsBySearchText = (userProductsBySearchText: Map<string, Map<string, any>>) => {
  return {
    type: ProductActionType.ADD_LIST_PRODUCT_SEARCH_TEXT,
    payload: { userProductsBySearchText }
  }
}

/**
 * Add a list of product by Category
 */
export const addProductsByCategory = (userProductsByCategory: Map<string, Map<string, any>>) => {
  return {
    type: ProductActionType.ADD_LIST_PRODUCT_CATEGORY,
    payload: { userProductsByCategory }
  }
}

/**
 * Add a list of product Attributes
 */
export const addProductsAttributes = (userProductsAttributes: Map<string, Map<string, any>>) => {
  return {
    type: ProductActionType.ADD_LIST_PRODUCT_ATTRIBUTES,
    payload: { userProductsAttributes }
  }
}

/**
 * Clea all data in product store
 */
export const clearAllData = () => {
  return {
    type: ProductActionType.CLEAR_ALL_DATA_PRODUCT
  }
}

/**
 * Clea search list in product store
 */
export const dbClearAllSearchList = () => {
  return {
    type: ProductActionType.CLEAR_SEARCH_DATA_PRODUCT
  }
}

/**
 * Clea category list in product store
 */
export const dbClearAllCategoryProductList = () => {
  return {
    type: ProductActionType.CLEAR_CATEGORY_DATA_PRODUCT
  }
}

/**
 * Add a product with image
 */
export const addImageProduct = (uid: string, product: any) => {
  return {
    type: ProductActionType.ADD_IMAGE_PRODUCT,
    payload: { uid, product }
  }

}

/**
 * Set stream has more data to show
 */
export const hasMoreDataStream = () => {
  return {
    type: ProductActionType.HAS_MORE_DATA_STREAM
  }

}

/**
 * Set stream has not data any more to show
 */
export const notMoreDataStream = () => {
  return {
    type: ProductActionType.NOT_MORE_DATA_STREAM
  }

}

/**
 * Set last page request of stream
 */
export const requestPageStream = (page: number) => {
  return {
    type: ProductActionType.REQUEST_PAGE_STREAM,
    payload: { page }
  }

}

/**
 * Set last product identification of stream
 */
export const lastProductStream = (lastProductId: string) => {
  return {
    type: ProductActionType.LAST_PRODUCT_STREAM,
    payload: { lastProductId }
  }

}

/**
 * Set profile products has more data to show
 */
export const hasMoreDataProfile = () => {
  return {
    type: ProductActionType.HAS_MORE_DATA_PROFILE
  }

}

/**
 * Set profile products has not data any more to show
 */
export const notMoreDataProfile = (userId: string) => {
  return {
    type: ProductActionType.NOT_MORE_DATA_PROFILE,
    payload: { userId }
  }

}

/**
 * Set last page request of profile products
 */
export const requestPageProfile = (userId: string, page: number) => {
  return {
    type: ProductActionType.REQUEST_PAGE_PROFILE,
    payload: { userId, page }
  }

}

/**
 * Set last product identification of profile products
 */
export const lastProductProfile = (userId: string, lastProductId: string) => {
  return {
    type: ProductActionType.LAST_PRODUCT_PROFILE,
    payload: { userId, lastProductId }
  }

}
