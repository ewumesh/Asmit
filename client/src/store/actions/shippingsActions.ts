// - Import domain
import { Map, fromJS} from 'immutable'

import { IShippingService } from 'src/core/services/shippings'
import { ShippingAddress } from 'core/domain/shippings'
import { ShippingsActionType } from 'constants/shippingsActionType'
import { SocialError } from 'src/core/domain/common'
import { SocialProviderTypes } from 'src/core/socialProviderTypes'
import { provider } from 'src/ecommerceEngine'
import * as checkoutActions from 'store/actions/checkoutActions'
import * as globalActions from 'store/actions/globalActions'

/**
 * Get service providers
 */
const shippingService: IShippingService = provider.get<IShippingService>(SocialProviderTypes.ShippingService)

/**
 * Get all user shipping rigions from data base
 */
export const dbGetShippingRegions = () => {
  return (dispatch: any, getState: Function) => {
    return shippingService.getShippingRegions().then((result) => {
      let parsedData: Map<string, Map<string, any>> = Map({})
      result.shippingRegions.forEach((shippingRegions) => {
        const shippingRegionsId = Object.keys(shippingRegions)[0]
        const shippingRegionsData = shippingRegions[shippingRegionsId]
        parsedData = parsedData.setIn([shippingRegionsId], fromJS(shippingRegionsData))
      })

      dispatch(addRegions(parsedData))
    })
    .catch((error: SocialError) => {
        dispatch(globalActions.showMessage(error.message))
    })
  }
}

export const dbAddShippingAddress = (shippingAddress: ShippingAddress) => {
  return (dispatch: any, getState: Function) => {
    dispatch(globalActions.showTopLoading())
    return shippingService.addShippingAddress(shippingAddress).then((result: ShippingAddress) => {
      dispatch(addShippingAddress(result))
      dispatch(globalActions.hideTopLoading())
    })
  }
}

export const dbAddShippingAddressfromcheckout = (shippingAddress: ShippingAddress) => {
  return (dispatch: any, getState: Function) => {
    dispatch(globalActions.showTopLoading())
    return shippingService.addShippingAddress(shippingAddress).then((result: ShippingAddress) => {
      dispatch(checkoutActions.setActiveStep(1))
      dispatch(addShippingAddress(result))
      dispatch(globalActions.hideTopLoading())
    })
  }
}

export const dbGetShippingAddress = () => {
  return (dispatch: any, getState: Function) => {
    dispatch(globalActions.showTopLoading())
    return shippingService.getShippingAddress().then((result: ShippingAddress) => {
      dispatch(addShippingAddress(result))
      dispatch(globalActions.hideTopLoading())
    })
  }
}

export const dbGetShippingAddressRefresh = () => {
  return (dispatch: any, getState: Function) => {
    return shippingService.getShippingAddress().then((result: ShippingAddress) => {
      dispatch(addShippingAddress(result))
    })
  }
}

/**
 * Get all user shipping rates from data base
 */
export const dbGetShippingRates = (regionId: string) => {
  return (dispatch: any, getState: Function) => {
    return shippingService.getShippingRates(regionId).then((result) => {
      let parsedData: Map<string, Map<string, any>> = Map({})
      result.shippingRates.forEach((shippingRates) => {
        const shippingId = Object.keys(shippingRates)[0]
        const shippingRatesData = shippingRates[shippingId]
        parsedData = parsedData.setIn([shippingId], fromJS(shippingRatesData))
      })

      dispatch(addRates(parsedData))
    })
    .catch((error: SocialError) => {
        dispatch(globalActions.showMessage(error.message))
    })
  }
}

/* _____________ CRUD State _____________ */

/**
 * Add a list of rigions
 */
export const addRegions = (shippingRegions: Map<string, Map<string, any>>) => {
  return {
    type: ShippingsActionType.ADD_LIST_SHIPPING_RIGIONS,
    payload: {shippingRegions}
  }
}

/**
 * Add a Shipping Address
 */
export const addShippingAddress = (shippingAddress: ShippingAddress) => {
  return {
    type: ShippingsActionType.ADD_SHIPPING_ADDRESS,
    payload: { shippingAddress }
  }
}

/**
 * Add a list of shipping Rates
 */
export const addRates = (shippingRates: Map<string, Map<string, any>>) => {
  return {
    type: ShippingsActionType.ADD_LIST_SHIPPING_RATES,
    payload: {shippingRates}
  }
}
