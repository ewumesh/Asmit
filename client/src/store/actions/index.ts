import * as authorizeActions from './authorizeActions'
import * as globalActions from './globalActions'
import * as imageGalleryActions from './imageGalleryActions'
import * as userActions from './userActions'
import * as localeActions from './localeActions'
import * as serverActions from './serverActions'
import * as productActions from './productActions'
import * as shippingsActions from './shippingsActions'
import * as addToCartActions from './addToCartActions'

export {
  authorizeActions,
  globalActions,
  imageGalleryActions,
  userActions,
  localeActions,
  serverActions,
  productActions,
  shippingsActions,
  addToCartActions
}
