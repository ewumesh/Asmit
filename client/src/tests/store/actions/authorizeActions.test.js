import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import nock from 'nock'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

import { authorizeActions } from '../../../store/actions'

describe('async actions', () => {
  let store
  let userData = [
    {
      customeId: 1,
      name: 'Kapil Jain',
      email: 'kapil.jain@ipragmatech.com'
    }
  ]
  beforeEach(() => {
    store = mockStore({})
  })
  afterEach(() => {
    // clear all HTTP mocks after each test
    nock.cleanAll()
  })

  it('Creates LOGIN when login has been done', () => {
    // Simulate a successful response
    nock('http://localhost:3000')
      .defaultReplyHeaders(
        { 'access-control-allow-origin': '*' },
        { 'Access-Control-Allow-Credentials': true },
        { 'Content-Type': 'application/json' }
      )
      .post(
        '/api/login',
        JSON.stringify({
          email: 'kapil.jain40@ipragmatech.com',
          password: '123456'
        })
      ) // Route to catch and mock
      .reply(200, userData) // Mock reponse code and data

    // Dispatch action to fetch to-dos
    return store
      .dispatch(
        authorizeActions.dbLogin('kapil.jain40@ipragmatech.com', '123456')
      )
      .then(() => {
        // return of async actions
        expect(store.getActions()).toMatchSnapshot()
      })
  })

  it('Creates LOGIN when login with wrong Credentials', () => {
    // Simulate a successful response
    nock('http://localhost:3000')
      .defaultReplyHeaders(
        { 'access-control-allow-origin': '*' },
        { 'Access-Control-Allow-Credentials': true },
        { 'Content-Type': 'application/json' }
      )
      .post(
        '/api/login',
        JSON.stringify({
          email: 'kapil.jai@ipragmatech.com',
          password: '123456'
        })
      ) // Route to catch and mock
      .reply(200, userData) // Mock reponse code and data

    // Dispatch action to fetch to-dos
    return store
      .dispatch(
        authorizeActions.dbLogin('kapil.jai@ipragmatech.com', '123456')
      )
      .then(() => {
        // return of async actions
        expect(store.getActions()).toMatchSnapshot()
      })
  })

  it('Creates Signup when user has been done', () => {
    // Simulate a successful response
    nock('http://localhost:3000')
      .defaultReplyHeaders(
        { 'access-control-allow-origin': '*' },
        { 'Access-Control-Allow-Credentials': true },
        { 'Content-Type': 'application/json' }
      )
      .post(
        '/api/signup',
        JSON.stringify({
          name: 'Kapil Jain',
          email: 'kapil.jain45@ipragmatech.com',
          password: '123456'
        })
      ) // Route to catch and mock
      .reply(200, userData) // Mock reponse code and data

    // Dispatch action to fetch to-dos
    return store

      .dispatch(
        authorizeActions.dbSignup({
          fullName: 'Kapil Jain',
          email: 'kapil.jain45@ipragmatech.com',
          password: '123456'
        })
      )
      .then(() => {
        // return of async actions
        expect(store.getActions()).toMatchSnapshot()
      })
  })

  it('Signout the user', () => {
    // Simulate a successful response
    nock('http://localhost:3000')
      .defaultReplyHeaders(
        { 'access-control-allow-origin': '*' },
        { 'Access-Control-Allow-Credentials': true },
        { 'Content-Type': 'application/json' }
      )
      .post(
        '/api/signout'
      ) // Route to catch and mock
      .reply(200, userData) // Mock reponse code and data

    // Dispatch action to fetch to-dos
    return store

      .dispatch(
        authorizeActions.dbLogout()
      )
      .then(() => {
        // return of async actions
        expect(store.getActions()).toMatchSnapshot()
      })
  })

})
