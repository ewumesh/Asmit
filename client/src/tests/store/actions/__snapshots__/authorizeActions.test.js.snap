// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions Creates LOGIN when login has been done 1`] = `
Array [
  Object {
    "payload": Object {
      "authed": true,
      "displayName": "Kapil Jain",
      "email": "kapil.jain40@ipragmatech.com",
      "uid": 2,
    },
    "type": "LOGIN",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`async actions Creates LOGIN when login with wrong Credentials 1`] = `
Array [
  Object {
    "payload": "Please check your email and password again",
    "type": "SHOW_MESSAGE_GLOBAL",
  },
]
`;

exports[`async actions Creates Signup when user has been done 1`] = `
Array [
  Object {
    "payload": "Bad request",
    "type": "SHOW_MESSAGE_GLOBAL",
  },
]
`;

exports[`async actions Creates Signup when user has been done 2`] = `
Array [
  Object {
    "payload": "Bad request",
    "type": "SHOW_MESSAGE_GLOBAL",
  },
]
`;

exports[`async actions Signout the user 1`] = `
Array [
  Object {
    "payload": Object {
      "msg": "User logout successfully",
      "status": true,
    },
    "type": "LOGOUT",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/login",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;
